csi/controller_server.go

```diff
@@ -747,6 +748,11 @@ func (cs *ControllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 	if existVol, err = cs.apiClient.Volume.ActionExpand(existVol, &longhornclient.ExpandInput{
 		Size: strconv.FormatInt(requestedSize, 10),
 	}); err != nil {
+		// TODO: This manual error code parsing should be refactored once Longhorn API implements error code response
+		// https://github.com/longhorn/longhorn/issues/1875
+		if matched, _ := regexp.MatchString("cannot schedule .* more bytes to disk", err.Error()); matched {
+			return nil, status.Errorf(codes.OutOfRange, err.Error())
+		}
```

Comment:
@joshimoo  Can you take a look at this error code parsing as we discussed in the PR review meeting?

Reply:
I think this should be fine, maybe a little overkill with the regex but since it's one off why not :)